heron/spi/src/java/com/twitter/heron/spi/utils/ShellUtils.java

```diff
@@ -60,63 +60,80 @@ static String inputstreamToString(InputStream is) {
   }
 
   public static int runProcess(
-      boolean verbose, String cmdline, StringBuilder stdout, StringBuilder stderr) {
-    return runSyncProcess(verbose, false, splitTokens(cmdline), stdout, stderr, null);
+      String cmdline, StringBuilder stderr) {
+    return runSyncProcess(false, splitTokens(cmdline), stderr, null);
   }
 
   public static int runSyncProcess(
-      boolean verbose, boolean isInheritIO, String[] cmdline, StringBuilder stdout,
-      StringBuilder stderr, File workingDirectory) {
-    return runSyncProcess(isInheritIO, cmdline, stdout, stderr, workingDirectory,
+      boolean isInheritIO, String[] cmdline, StringBuilder stderr, File workingDirectory) {
+    return runSyncProcess(isInheritIO, cmdline, stderr, workingDirectory,
         new HashMap<String, String>());
   }
 
   /**
    * Start a daemon thread to read data from "input" to "out".
    */
-  private static Thread asyncProcessStream(final InputStream input, final StringBuilder out) {
+  private static Thread asyncProcessStream(final InputStream input, final StringBuilder builder) {
     Thread thread = new Thread() {
       @Override
       public void run() {
-        try {
-          out.append(inputstreamToString(input));
-        } finally {
+        // do not buffer
+        System.err.println("Process output (stdout+stderr):");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(input), 1);
+        while(true) {
+          String line = null;
           try {
-            input.close();
+            line = reader.readLine();
           } catch (IOException e) {
-            LOG.log(Level.WARNING, "Failed to close the input stream", e);
+            LOG.log(Level.SEVERE, "Error when reading line from subprocess", e);
+          }
+          if (line == null) {
+            break;
+          } else {
+            System.err.println(line);
+            builder.append(line);
           }
         }
+        try {
+          input.close();
+        } catch (IOException e) {
+          LOG.log(Level.WARNING, "Failed to close the input stream", e);
+        }
       }
     };
     thread.setDaemon(true);
-    thread.start();
     return thread;
   }
 
   /**
    * run sync process
    */
   private static int runSyncProcess(
-      boolean isInheritIO, String[] cmdline, StringBuilder stdout,
+      boolean isInheritIO, String[] cmdline,
       StringBuilder stderr, File workingDirectory, Map<String, String> envs) {
-    final StringBuilder pStdOut = stdout == null ? new StringBuilder() : stdout;
     final StringBuilder pStdErr = stderr == null ? new StringBuilder() : stderr;
 
     // Log the command for debugging
-    LOG.log(Level.FINE, "Process command: `$ {0}`", Arrays.toString(cmdline));
+    LOG.log(Level.INFO, "Running synced process:\n``{0}''''", String.join(" ", cmdline));
     ProcessBuilder pb = getProcessBuilder(isInheritIO, cmdline, workingDirectory, envs);
+    /* combine input stream and error stream because
```

Comment:
pls clarify that we're combining the two into stderr, which I think is the case.